src/vsphere_cpi/lib/cloud/vsphere/vm_type.rb

```diff
@@ -27,5 +27,13 @@ def nsx_security_groups
     def nsx_lbs
       nsx['lbs'] if nsx
     end
+
+    def nsxt_server_pools
+      nsxt['lb']['server_pools'] if nsxt && nsxt['lb']
+    end
+
+    def ns_groups
```

Comment:
Did we change this to `ns_groups` from `nsgroups`?

Reply:
in cloud config it is actually ns_group. One of the tests was not implemented right where the config was incorrect. Also, we reference it as ns_group and nsgroup at different places. I used what we had as per the config. Need to refactor overall to match..